addons/calendar/models/res_users_settings.py

```diff
@@ -0,0 +1,21 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class ResUsersSettings(models.Model):
+    _inherit = "res.users.settings"
+
+    # Calendar module settings.
+    calendar_default_privacy = fields.Selection(
+        [('public', 'Public'),
+         ('private', 'Private'),
+         ('confidential', 'Only internal users')],
+        'Calendar Default Privacy', default='public', required=True,
```

Comment:
```suggestion
        'Calendar Default Privacy', default='private', required=True,
```
privacy by default please

Reply:
> No, the events will follow the privacy option set in the external calendar (Google or Microsoft).

But the description of the task ("when a user synchronized events from Google/Microsoft calendar to Odoo, those events would be visible to everyone at first.") suggest it’s public by default. In which scenario does your PR makes a difference?

> newly created events will be private if the user dont explicitely defined a privacy option for them. Considering this, what do you think?

Yes, this is exactly what I am suggesting. I don’t understand why event are public by default, that sounds like something facebook would do, not a business software…